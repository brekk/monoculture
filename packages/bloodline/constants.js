export const DEFAULT_GRAPHVIZ_CONFIG = {
  shape: 'box',
  style: 'rounded',
  edgeColor: '#888888',
  fontsize: '16px',
  fontname: `Fira Code`,
  nodeColor: '#fff',
  rankdir: 'LR',
  layout: 'dot',
  bgcolor: '#111111',
  colors: {
    noDependency: '#00cc00',
    cyclical: '#cc0000',
  },
}
