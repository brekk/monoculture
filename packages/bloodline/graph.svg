<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
 "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Generated by graphviz version 9.0.0 (20230911.1827)
 -->
<!-- Title: G Pages: 1 -->
<svg width="641pt" height="351pt"
 viewBox="0.00 0.00 640.95 350.95" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(21.6 329.35)">
<title>G</title>
<polygon fill="#111111" stroke="none" points="-21.6,21.6 -21.6,-329.35 619.35,-329.35 619.35,21.6 -21.6,21.6"/>
<!-- fluture -->
<g id="node1" class="node">
<title>fluture</title>
<path fill="none" stroke="#00cc00" d="M228.71,-307.75C228.71,-307.75 164.29,-307.75 164.29,-307.75 159.33,-307.75 154.38,-302.79 154.38,-297.83 154.38,-297.83 154.38,-287.92 154.38,-287.92 154.38,-282.96 159.33,-278 164.29,-278 164.29,-278 228.71,-278 228.71,-278 233.67,-278 238.62,-282.96 238.62,-287.92 238.62,-287.92 238.62,-297.83 238.62,-297.83 238.62,-302.79 233.67,-307.75 228.71,-307.75"/>
<text text-anchor="middle" x="196.5" y="-288.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">fluture</text>
</g>
<!-- ramda -->
<g id="node2" class="node">
<title>ramda</title>
<path fill="none" stroke="#00cc00" d="M470.83,-269.75C470.83,-269.75 425.92,-269.75 425.92,-269.75 420.96,-269.75 416,-264.79 416,-259.83 416,-259.83 416,-249.92 416,-249.92 416,-244.96 420.96,-240 425.92,-240 425.92,-240 470.83,-240 470.83,-240 475.79,-240 480.75,-244.96 480.75,-249.92 480.75,-249.92 480.75,-259.83 480.75,-259.83 480.75,-264.79 475.79,-269.75 470.83,-269.75"/>
<text text-anchor="middle" x="448.38" y="-250.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">ramda</text>
</g>
<!-- config.js -->
<g id="node3" class="node">
<title>config.js</title>
<path fill="none" stroke="#00cc00" d="M238.46,-221.75C238.46,-221.75 154.54,-221.75 154.54,-221.75 149.58,-221.75 144.62,-216.79 144.62,-211.83 144.62,-211.83 144.62,-201.92 144.62,-201.92 144.62,-196.96 149.58,-192 154.54,-192 154.54,-192 238.46,-192 238.46,-192 243.42,-192 248.38,-196.96 248.38,-201.92 248.38,-201.92 248.38,-211.83 248.38,-211.83 248.38,-216.79 243.42,-221.75 238.46,-221.75"/>
<text text-anchor="middle" x="196.5" y="-202.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">config.js</text>
</g>
<!-- bloodline.js -->
<g id="node4" class="node">
<title>bloodline.js</title>
<path fill="none" stroke="#ffffff" d="M253.08,-173.75C253.08,-173.75 139.92,-173.75 139.92,-173.75 134.96,-173.75 130,-168.79 130,-163.83 130,-163.83 130,-153.92 130,-153.92 130,-148.96 134.96,-144 139.92,-144 139.92,-144 253.08,-144 253.08,-144 258.04,-144 263,-148.96 263,-153.92 263,-153.92 263,-163.83 263,-163.83 263,-168.79 258.04,-173.75 253.08,-173.75"/>
<text text-anchor="middle" x="196.5" y="-154.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">bloodline.js</text>
</g>
<!-- bloodline.js&#45;&gt;ramda -->
<g id="edge1" class="edge">
<title>bloodline.js&#45;&gt;ramda</title>
<path fill="none" stroke="#888888" d="M248.39,-174.12C253.55,-176.65 258.53,-179.55 263,-182.88 284.4,-198.78 276.18,-217.08 299,-230.88 330.68,-250.03 372.82,-255.33 404.09,-256.24"/>
<polygon fill="#888888" stroke="#888888" points="403.94,-259.74 413.99,-256.37 404.04,-252.74 403.94,-259.74"/>
</g>
<!-- ../file&#45;system/file&#45;system.js -->
<g id="node5" class="node">
<title>../file&#45;system/file&#45;system.js</title>
<path fill="none" stroke="#00cc00" d="M587.83,-221.75C587.83,-221.75 308.92,-221.75 308.92,-221.75 303.96,-221.75 299,-216.79 299,-211.83 299,-211.83 299,-201.92 299,-201.92 299,-196.96 303.96,-192 308.92,-192 308.92,-192 587.83,-192 587.83,-192 592.79,-192 597.75,-196.96 597.75,-201.92 597.75,-201.92 597.75,-211.83 597.75,-211.83 597.75,-216.79 592.79,-221.75 587.83,-221.75"/>
<text text-anchor="middle" x="448.38" y="-202.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">../file&#45;system/file&#45;system.js</text>
</g>
<!-- bloodline.js&#45;&gt;../file&#45;system/file&#45;system.js -->
<g id="edge2" class="edge">
<title>bloodline.js&#45;&gt;../file&#45;system/file&#45;system.js</title>
<path fill="none" stroke="#888888" d="M259.34,-174.24C272.4,-177.28 286.12,-180.31 299,-182.88 310.66,-185.19 322.87,-187.45 335.06,-189.58"/>
<polygon fill="#888888" stroke="#888888" points="334.29,-193 344.74,-191.26 335.49,-186.11 334.29,-193"/>
</g>
<!-- constants.js -->
<g id="node6" class="node">
<title>constants.js</title>
<path fill="none" stroke="#00cc00" d="M504.96,-173.75C504.96,-173.75 391.79,-173.75 391.79,-173.75 386.83,-173.75 381.88,-168.79 381.88,-163.83 381.88,-163.83 381.88,-153.92 381.88,-153.92 381.88,-148.96 386.83,-144 391.79,-144 391.79,-144 504.96,-144 504.96,-144 509.92,-144 514.88,-148.96 514.88,-153.92 514.88,-153.92 514.88,-163.83 514.88,-163.83 514.88,-168.79 509.92,-173.75 504.96,-173.75"/>
<text text-anchor="middle" x="448.38" y="-154.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">constants.js</text>
</g>
<!-- bloodline.js&#45;&gt;constants.js -->
<g id="edge3" class="edge">
<title>bloodline.js&#45;&gt;constants.js</title>
<path fill="none" stroke="#888888" d="M263.48,-158.88C296.21,-158.88 335.96,-158.88 370.11,-158.88"/>
<polygon fill="#888888" stroke="#888888" points="369.95,-162.38 379.95,-158.88 369.95,-155.38 369.95,-162.38"/>
</g>
<!-- tree.js -->
<g id="node7" class="node">
<title>tree.js</title>
<path fill="none" stroke="#00cc00" d="M480.58,-125.75C480.58,-125.75 416.17,-125.75 416.17,-125.75 411.21,-125.75 406.25,-120.79 406.25,-115.83 406.25,-115.83 406.25,-105.92 406.25,-105.92 406.25,-100.96 411.21,-96 416.17,-96 416.17,-96 480.58,-96 480.58,-96 485.54,-96 490.5,-100.96 490.5,-105.92 490.5,-105.92 490.5,-115.83 490.5,-115.83 490.5,-120.79 485.54,-125.75 480.58,-125.75"/>
<text text-anchor="middle" x="448.38" y="-106.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">tree.js</text>
</g>
<!-- bloodline.js&#45;&gt;tree.js -->
<g id="edge4" class="edge">
<title>bloodline.js&#45;&gt;tree.js</title>
<path fill="none" stroke="#888888" d="M259.34,-143.51C272.4,-140.47 286.12,-137.44 299,-134.88 330.6,-128.59 366.24,-122.79 394.67,-118.46"/>
<polygon fill="#888888" stroke="#888888" points="395.11,-121.93 404.48,-116.98 394.07,-115.01 395.11,-121.93"/>
</g>
<!-- visualization.js -->
<g id="node8" class="node">
<title>visualization.js</title>
<path fill="none" stroke="#00cc00" d="M524.46,-77.75C524.46,-77.75 372.29,-77.75 372.29,-77.75 367.33,-77.75 362.38,-72.79 362.38,-67.83 362.38,-67.83 362.38,-57.92 362.38,-57.92 362.38,-52.96 367.33,-48 372.29,-48 372.29,-48 524.46,-48 524.46,-48 529.42,-48 534.38,-52.96 534.38,-57.92 534.38,-57.92 534.38,-67.83 534.38,-67.83 534.38,-72.79 529.42,-77.75 524.46,-77.75"/>
<text text-anchor="middle" x="448.38" y="-58.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">visualization.js</text>
</g>
<!-- bloodline.js&#45;&gt;visualization.js -->
<g id="edge5" class="edge">
<title>bloodline.js&#45;&gt;visualization.js</title>
<path fill="none" stroke="#888888" d="M248.39,-143.63C253.55,-141.1 258.53,-138.2 263,-134.88 284.4,-118.97 276.18,-100.67 299,-86.88 314.63,-77.43 332.8,-71.35 350.95,-67.51"/>
<polygon fill="#888888" stroke="#888888" points="351.56,-70.96 360.74,-65.67 350.27,-64.08 351.56,-70.96"/>
</g>
<!-- log.js -->
<g id="node9" class="node">
<title>log.js</title>
<path fill="none" stroke="#00cc00" d="M475.71,-29.75C475.71,-29.75 421.04,-29.75 421.04,-29.75 416.08,-29.75 411.12,-24.79 411.12,-19.83 411.12,-19.83 411.12,-9.92 411.12,-9.92 411.12,-4.96 416.08,0 421.04,0 421.04,0 475.71,0 475.71,0 480.67,0 485.62,-4.96 485.62,-9.92 485.62,-9.92 485.62,-19.83 485.62,-19.83 485.62,-24.79 480.67,-29.75 475.71,-29.75"/>
<text text-anchor="middle" x="448.38" y="-10.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">log.js</text>
</g>
<!-- bloodline.js&#45;&gt;log.js -->
<g id="edge6" class="edge">
<title>bloodline.js&#45;&gt;log.js</title>
<path fill="none" stroke="#888888" d="M251.68,-143.55C255.8,-141.07 259.65,-138.2 263,-134.88 295.34,-102.78 263.75,-67.75 299,-38.88 326.87,-16.05 367.83,-10.86 399.65,-10.93"/>
<polygon fill="#888888" stroke="#888888" points="399.34,-14.42 409.41,-11.14 399.49,-7.42 399.34,-14.42"/>
</g>
<!-- package.json -->
<g id="node10" class="node">
<title>package.json</title>
<path fill="none" stroke="#00cc00" d="M253.08,-125.75C253.08,-125.75 139.92,-125.75 139.92,-125.75 134.96,-125.75 130,-120.79 130,-115.83 130,-115.83 130,-105.92 130,-105.92 130,-100.96 134.96,-96 139.92,-96 139.92,-96 253.08,-96 253.08,-96 258.04,-96 263,-100.96 263,-105.92 263,-105.92 263,-115.83 263,-115.83 263,-120.79 258.04,-125.75 253.08,-125.75"/>
<text text-anchor="middle" x="196.5" y="-106.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">package.json</text>
</g>
<!-- ./cli.js -->
<g id="node11" class="node">
<title>./cli.js</title>
<path fill="none" stroke="#ffffff" d="M84.08,-221.75C84.08,-221.75 9.92,-221.75 9.92,-221.75 4.96,-221.75 0,-216.79 0,-211.83 0,-211.83 0,-201.92 0,-201.92 0,-196.96 4.96,-192 9.92,-192 9.92,-192 84.08,-192 84.08,-192 89.04,-192 94,-196.96 94,-201.92 94,-201.92 94,-211.83 94,-211.83 94,-216.79 89.04,-221.75 84.08,-221.75"/>
<text text-anchor="middle" x="47" y="-202.55" font-family="Fira Code" font-size="16.00" fill="#ffffff">./cli.js</text>
</g>
<!-- ./cli.js&#45;&gt;fluture -->
<g id="edge7" class="edge">
<title>./cli.js&#45;&gt;fluture</title>
<path fill="none" stroke="#888888" d="M67.53,-222.04C83.72,-234.21 107.63,-251.3 130,-263.88 135.42,-266.92 141.22,-269.89 147.05,-272.68"/>
<polygon fill="#888888" stroke="#888888" points="145.57,-275.85 156.12,-276.88 148.51,-269.5 145.57,-275.85"/>
</g>
<!-- ./cli.js&#45;&gt;ramda -->
<g id="edge8" class="edge">
<title>./cli.js&#45;&gt;ramda</title>
<path fill="none" stroke="#888888" d="M94,-222.17C105.63,-225.53 118.18,-228.72 130,-230.88 227.04,-248.54 343.05,-253.24 404.4,-254.47"/>
<polygon fill="#888888" stroke="#888888" points="403.99,-257.96 414.05,-254.64 404.11,-250.96 403.99,-257.96"/>
</g>
<!-- ./cli.js&#45;&gt;config.js -->
<g id="edge9" class="edge">
<title>./cli.js&#45;&gt;config.js</title>
<path fill="none" stroke="#888888" d="M94.27,-206.88C106.55,-206.88 120.02,-206.88 133.02,-206.88"/>
<polygon fill="#888888" stroke="#888888" points="132.76,-210.38 142.76,-206.88 132.76,-203.38 132.76,-210.38"/>
</g>
<!-- ./cli.js&#45;&gt;bloodline.js -->
<g id="edge10" class="edge">
<title>./cli.js&#45;&gt;bloodline.js</title>
<path fill="none" stroke="#888888" d="M94.27,-191.82C108,-187.35 123.22,-182.4 137.59,-177.72"/>
<polygon fill="#888888" stroke="#888888" points="138.52,-181.1 146.94,-174.68 136.35,-174.44 138.52,-181.1"/>
</g>
<!-- ./cli.js&#45;&gt;package.json -->
<g id="edge11" class="edge">
<title>./cli.js&#45;&gt;package.json</title>
<path fill="none" stroke="#888888" d="M62.09,-191.52C77.46,-175.52 103.45,-150.64 130,-134.88 132.28,-133.52 134.66,-132.22 137.09,-130.97"/>
<polygon fill="#888888" stroke="#888888" points="138.38,-134.23 145.95,-126.82 135.41,-127.89 138.38,-134.23"/>
</g>
</g>
</svg>
