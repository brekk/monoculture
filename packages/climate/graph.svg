<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
 "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Generated by graphviz version 9.0.0 (20230911.1827)
 -->
<!-- Title: G Pages: 1 -->
<svg width="324pt" height="151pt"
 viewBox="0.00 0.00 324.45 150.95" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(21.6 129.35)">
<title>G</title>
<polygon fill="#111111" stroke="none" points="-21.6,21.6 -21.6,-129.35 302.85,-129.35 302.85,21.6 -21.6,21.6"/>
<!-- builder.js -->
<g id="node1" class="node">
<title>builder.js</title>
<path fill="none" stroke="#c6c5fe" d="M170.33,-66.75C170.33,-66.75 116.92,-66.75 116.92,-66.75 112.96,-66.75 109,-62.79 109,-58.83 109,-58.83 109,-50.92 109,-50.92 109,-46.96 112.96,-43 116.92,-43 116.92,-43 170.33,-43 170.33,-43 174.29,-43 178.25,-46.96 178.25,-50.92 178.25,-50.92 178.25,-58.83 178.25,-58.83 178.25,-62.79 174.29,-66.75 170.33,-66.75"/>
<text text-anchor="middle" x="143.62" y="-49.45" font-family="Arial" font-size="14.00" fill="#c6c5fe">builder.js</text>
</g>
<!-- help.js -->
<g id="node2" class="node">
<title>help.js</title>
<path fill="none" stroke="#cfffac" d="M267.33,-107.75C267.33,-107.75 228.17,-107.75 228.17,-107.75 224.21,-107.75 220.25,-103.79 220.25,-99.83 220.25,-99.83 220.25,-91.92 220.25,-91.92 220.25,-87.96 224.21,-84 228.17,-84 228.17,-84 267.33,-84 267.33,-84 271.29,-84 275.25,-87.96 275.25,-91.92 275.25,-91.92 275.25,-99.83 275.25,-99.83 275.25,-103.79 271.29,-107.75 267.33,-107.75"/>
<text text-anchor="middle" x="247.75" y="-90.45" font-family="Arial" font-size="14.00" fill="#cfffac">help.js</text>
</g>
<!-- builder.js&#45;&gt;help.js -->
<g id="edge1" class="edge">
<title>builder.js&#45;&gt;help.js</title>
<path fill="none" stroke="#757575" d="M175.27,-67.18C186.04,-71.5 198.27,-76.41 209.52,-80.93"/>
<polygon fill="#757575" stroke="#757575" points="207.95,-84.07 218.54,-84.55 210.56,-77.57 207.95,-84.07"/>
</g>
<!-- log.js -->
<g id="node3" class="node">
<title>log.js</title>
<path fill="none" stroke="#cfffac" d="M266.83,-65.75C266.83,-65.75 228.67,-65.75 228.67,-65.75 224.71,-65.75 220.75,-61.79 220.75,-57.83 220.75,-57.83 220.75,-49.92 220.75,-49.92 220.75,-45.96 224.71,-42 228.67,-42 228.67,-42 266.83,-42 266.83,-42 270.79,-42 274.75,-45.96 274.75,-49.92 274.75,-49.92 274.75,-57.83 274.75,-57.83 274.75,-61.79 270.79,-65.75 266.83,-65.75"/>
<text text-anchor="middle" x="247.75" y="-48.45" font-family="Arial" font-size="14.00" fill="#cfffac">log.js</text>
</g>
<!-- builder.js&#45;&gt;log.js -->
<g id="edge2" class="edge">
<title>builder.js&#45;&gt;log.js</title>
<path fill="none" stroke="#757575" d="M178.63,-54.54C188.39,-54.45 199.09,-54.34 209.05,-54.24"/>
<polygon fill="#757575" stroke="#757575" points="208.84,-57.75 218.81,-54.15 208.77,-50.75 208.84,-57.75"/>
</g>
<!-- parser.js -->
<g id="node4" class="node">
<title>parser.js</title>
<path fill="none" stroke="#cfffac" d="M273.33,-23.75C273.33,-23.75 222.17,-23.75 222.17,-23.75 218.21,-23.75 214.25,-19.79 214.25,-15.83 214.25,-15.83 214.25,-7.92 214.25,-7.92 214.25,-3.96 218.21,0 222.17,0 222.17,0 273.33,0 273.33,0 277.29,0 281.25,-3.96 281.25,-7.92 281.25,-7.92 281.25,-15.83 281.25,-15.83 281.25,-19.79 277.29,-23.75 273.33,-23.75"/>
<text text-anchor="middle" x="247.75" y="-6.45" font-family="Arial" font-size="14.00" fill="#cfffac">parser.js</text>
</g>
<!-- builder.js&#45;&gt;parser.js -->
<g id="edge3" class="edge">
<title>builder.js&#45;&gt;parser.js</title>
<path fill="none" stroke="#757575" d="M173.89,-42.55C184.25,-38.19 196.09,-33.21 207.17,-28.54"/>
<polygon fill="#757575" stroke="#757575" points="208.23,-31.89 216.09,-24.79 205.51,-25.44 208.23,-31.89"/>
</g>
<!-- climate.js -->
<g id="node5" class="node">
<title>climate.js</title>
<path fill="none" stroke="#c6c5fe" d="M65.08,-86.75C65.08,-86.75 7.92,-86.75 7.92,-86.75 3.96,-86.75 0,-82.79 0,-78.83 0,-78.83 0,-70.92 0,-70.92 0,-66.96 3.96,-63 7.92,-63 7.92,-63 65.08,-63 65.08,-63 69.04,-63 73,-66.96 73,-70.92 73,-70.92 73,-78.83 73,-78.83 73,-82.79 69.04,-86.75 65.08,-86.75"/>
<text text-anchor="middle" x="36.5" y="-69.45" font-family="Arial" font-size="14.00" fill="#c6c5fe">climate.js</text>
</g>
<!-- climate.js&#45;&gt;builder.js -->
<g id="edge4" class="edge">
<title>climate.js&#45;&gt;builder.js</title>
<path fill="none" stroke="#757575" d="M73.38,-68.05C81.15,-66.57 89.45,-64.99 97.51,-63.46"/>
<polygon fill="#757575" stroke="#757575" points="98.02,-66.92 107.19,-61.62 96.71,-60.05 98.02,-66.92"/>
</g>
<!-- climate.js&#45;&gt;help.js -->
<g id="edge5" class="edge">
<title>climate.js&#45;&gt;help.js</title>
<path fill="none" stroke="#757575" d="M73.47,-78.49C111.17,-82.27 170.16,-88.19 208.64,-92.05"/>
<polygon fill="#757575" stroke="#757575" points="208.07,-95.51 218.37,-93.03 208.77,-88.55 208.07,-95.51"/>
</g>
</g>
</svg>
