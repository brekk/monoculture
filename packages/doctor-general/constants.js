export const MAGIC_IMPORT_KEY = 'drgen-import-above'
export const TESTABLE_EXAMPLE = 'test=true'
