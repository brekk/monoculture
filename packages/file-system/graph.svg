<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
 "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Generated by graphviz version 9.0.0 (20230911.1827)
 -->
<!-- Title: G Pages: 1 -->
<svg width="220pt" height="211pt"
 viewBox="0.00 0.00 219.95 210.95" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(21.6 189.35)">
<title>G</title>
<polygon fill="#111111" stroke="none" points="-21.6,21.6 -21.6,-189.35 198.35,-189.35 198.35,21.6 -21.6,21.6"/>
<!-- find&#45;up.js -->
<g id="node1" class="node">
<title>find&#45;up.js</title>
<path fill="none" stroke="#ff6c60" d="M164.71,-149.75C164.71,-149.75 109.79,-149.75 109.79,-149.75 105.83,-149.75 101.88,-145.79 101.88,-141.83 101.88,-141.83 101.88,-133.92 101.88,-133.92 101.88,-129.96 105.83,-126 109.79,-126 109.79,-126 164.71,-126 164.71,-126 168.67,-126 172.62,-129.96 172.62,-133.92 172.62,-133.92 172.62,-141.83 172.62,-141.83 172.62,-145.79 168.67,-149.75 164.71,-149.75"/>
<text text-anchor="middle" x="137.25" y="-132.45" font-family="Arial" font-size="14.00" fill="#ff6c60">find&#45;up.js</text>
</g>
<!-- find&#45;up.js&#45;&gt;find&#45;up.js -->
<g id="edge1" class="edge">
<title>find&#45;up.js&#45;&gt;find&#45;up.js</title>
<path fill="none" stroke="#757575" d="M119.45,-150.15C113.66,-158.97 119.59,-167.75 137.25,-167.75 146.63,-167.75 152.7,-165.27 155.47,-161.63"/>
<polygon fill="#757575" stroke="#757575" points="158.97,-161.53 155.1,-151.66 151.97,-161.78 158.97,-161.53"/>
</g>
<!-- fs.js -->
<g id="node2" class="node">
<title>fs.js</title>
<path fill="none" stroke="#cfffac" d="M156.33,-107.75C156.33,-107.75 118.17,-107.75 118.17,-107.75 114.21,-107.75 110.25,-103.79 110.25,-99.83 110.25,-99.83 110.25,-91.92 110.25,-91.92 110.25,-87.96 114.21,-84 118.17,-84 118.17,-84 156.33,-84 156.33,-84 160.29,-84 164.25,-87.96 164.25,-91.92 164.25,-91.92 164.25,-99.83 164.25,-99.83 164.25,-103.79 160.29,-107.75 156.33,-107.75"/>
<text text-anchor="middle" x="137.25" y="-90.45" font-family="Arial" font-size="14.00" fill="#cfffac">fs.js</text>
</g>
<!-- index.js -->
<g id="node3" class="node">
<title>index.js</title>
<path fill="none" stroke="#c6c5fe" d="M53.83,-86.75C53.83,-86.75 7.92,-86.75 7.92,-86.75 3.96,-86.75 0,-82.79 0,-78.83 0,-78.83 0,-70.92 0,-70.92 0,-66.96 3.96,-63 7.92,-63 7.92,-63 53.83,-63 53.83,-63 57.79,-63 61.75,-66.96 61.75,-70.92 61.75,-70.92 61.75,-78.83 61.75,-78.83 61.75,-82.79 57.79,-86.75 53.83,-86.75"/>
<text text-anchor="middle" x="30.88" y="-69.45" font-family="Arial" font-size="14.00" fill="#c6c5fe">index.js</text>
</g>
<!-- index.js&#45;&gt;find&#45;up.js -->
<g id="edge2" class="edge">
<title>index.js&#45;&gt;find&#45;up.js</title>
<path fill="none" stroke="#757575" d="M50.4,-87.17C63.49,-95.73 81.52,-107.29 97.75,-116.88 99.62,-117.98 101.55,-119.1 103.5,-120.21"/>
<polygon fill="#757575" stroke="#757575" points="101.55,-123.13 111.98,-124.95 104.96,-117.02 101.55,-123.13"/>
</g>
<!-- index.js&#45;&gt;fs.js -->
<g id="edge3" class="edge">
<title>index.js&#45;&gt;fs.js</title>
<path fill="none" stroke="#757575" d="M62.06,-80.95C73.56,-83.26 86.8,-85.93 98.89,-88.36"/>
<polygon fill="#757575" stroke="#757575" points="98.05,-91.76 108.54,-90.3 99.43,-84.9 98.05,-91.76"/>
</g>
<!-- interpret.js -->
<g id="node4" class="node">
<title>interpret.js</title>
<path fill="none" stroke="#cfffac" d="M168.83,-65.75C168.83,-65.75 105.67,-65.75 105.67,-65.75 101.71,-65.75 97.75,-61.79 97.75,-57.83 97.75,-57.83 97.75,-49.92 97.75,-49.92 97.75,-45.96 101.71,-42 105.67,-42 105.67,-42 168.83,-42 168.83,-42 172.79,-42 176.75,-45.96 176.75,-49.92 176.75,-49.92 176.75,-57.83 176.75,-57.83 176.75,-61.79 172.79,-65.75 168.83,-65.75"/>
<text text-anchor="middle" x="137.25" y="-48.45" font-family="Arial" font-size="14.00" fill="#cfffac">interpret.js</text>
</g>
<!-- index.js&#45;&gt;interpret.js -->
<g id="edge4" class="edge">
<title>index.js&#45;&gt;interpret.js</title>
<path fill="none" stroke="#757575" d="M62.06,-68.8C69.68,-67.27 78.06,-65.58 86.37,-63.91"/>
<polygon fill="#757575" stroke="#757575" points="86.96,-67.36 96.08,-61.96 85.58,-60.5 86.96,-67.36"/>
</g>
<!-- path.js -->
<g id="node5" class="node">
<title>path.js</title>
<path fill="none" stroke="#cfffac" d="M157.21,-23.75C157.21,-23.75 117.29,-23.75 117.29,-23.75 113.33,-23.75 109.38,-19.79 109.38,-15.83 109.38,-15.83 109.38,-7.92 109.38,-7.92 109.38,-3.96 113.33,0 117.29,0 117.29,0 157.21,0 157.21,0 161.17,0 165.12,-3.96 165.12,-7.92 165.12,-7.92 165.12,-15.83 165.12,-15.83 165.12,-19.79 161.17,-23.75 157.21,-23.75"/>
<text text-anchor="middle" x="137.25" y="-6.45" font-family="Arial" font-size="14.00" fill="#cfffac">path.js</text>
</g>
<!-- index.js&#45;&gt;path.js -->
<g id="edge5" class="edge">
<title>index.js&#45;&gt;path.js</title>
<path fill="none" stroke="#757575" d="M50.4,-62.58C63.49,-54.02 81.52,-42.46 97.75,-32.88 99.62,-31.77 101.55,-30.65 103.5,-29.54"/>
<polygon fill="#757575" stroke="#757575" points="104.96,-32.73 111.98,-24.8 101.55,-26.62 104.96,-32.73"/>
</g>
</g>
</svg>
