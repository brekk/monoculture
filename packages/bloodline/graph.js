import { always as K, pipe, curry } from 'ramda'
import { execWithCancel } from 'file-system'
import { coalesce } from 'fluture'
import { join as sysPathJoin } from 'node:path'

//
export const checkForGraphvizWithCancel = curry((cancel, pathToLook) =>
  pipe(
    execWithCancel(
      cancel,
      pathToLook ? sysPathJoin(pathToLook, 'gvpr') : 'gvpr'
    ),
    coalesce(K(false))(K(true))
  )(['-V'])
)
export const checkForGraphvizAtPath = checkForGraphvizWithCancel(() => {})
export const checkForGraphviz = () => checkForGraphvizAtPath('')
