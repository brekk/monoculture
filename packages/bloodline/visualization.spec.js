import { fork } from 'fluture'
import {
  dotStreamAdapterWithCancel,
  isNotEmpty,
  setAttribute,
  setColor,
  getCyclic,
  createGraph,
  generateOptions,
  createSVG,
} from './visualization'
import { checkForGraphviz } from './graph'
import { DEFAULT_GRAPHVIZ_CONFIG } from './constants'

test('dotStreamAdapterWithCancel', () => {
  expect(dotStreamAdapterWithCancel).toBeTruthy()
})
test('isNotEmpty', () => {
  expect(isNotEmpty).toBeTruthy()
})
test('setAttribute', () => {
  expect(setAttribute).toBeTruthy()
})
test('setColor', () => {
  expect(setColor).toBeTruthy()
})
test('getCyclic', () => {
  expect(getCyclic).toBeTruthy()
})
test('createGraph', () => {
  expect(createGraph).toBeTruthy()
})
test('generateOptions', () => {
  expect(generateOptions).toBeTruthy()
  expect(generateOptions({})).toMatchSnapshot()
})
test('createSVG', done => {
  expect(createSVG).toBeTruthy()
  fork(() => {
    expect('skipping tests which rely upon `gvpr`').toBeTruthy()
    done()
  })(() => {
    const modules = { 'a.js': ['b.js', 'c.js'], 'b.js': ['c.js'], 'c.js': [] }
    const cancel = () => {}
    fork(done)(z => {
      expect(z.toString()).toMatchSnapshot()
      done()
    })(createSVG(cancel, DEFAULT_GRAPHVIZ_CONFIG, [], modules))
  })(checkForGraphviz())
})
