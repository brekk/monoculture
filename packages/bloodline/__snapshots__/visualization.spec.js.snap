// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateOptions 1`] = `
{
  "attributes": {
    "edge": {
      "color": "#888888",
    },
    "graph": {
      "bgcolor": "#111111",
      "layout": "dot",
      "overlap": false,
      "pad": 0.3,
      "rankdir": "LR",
    },
    "node": {
      "color": "#fff",
      "fontcolor": "#fff",
      "fontname": "Fira Code",
      "fontsize": "16px",
      "height": 0,
      "shape": "box",
      "style": "rounded",
    },
  },
  "dotCommand": undefined,
}
`;
